src/Basis/Constructors.jl

```diff
@@ -2,27 +2,182 @@
+sausagetransformation(d)
+```
+
+# Arguments:
+- `d`: polynomial degree of truncated Taylor series expansion of arcsin(s).
+
+# Returns:
+Conformal mapping of Gauss ellipses to sausagetransformations using a
+truncated Taylor expansion of arcsin(s). See Figure 4.1 of Hale and Trefethen (2008).
+
+# Example:
+```jldoctest
+# sausagetransformation conformal map
+g, gprime = LFAToolkit.sausagetransformation(9)
+
+# verify
+@assert g.(LinRange(-1,1,5)) ≈ [-0.9999999999999999, -0.39765215163451934, 0.0, 0.39765215163451934, 0.9999999999999999]
+
+# output
+
+```
+"""
+function sausagetransformation(d)
+    c = zeros(d + 1)
+    c[2:2:end] = [1, cumprod(1:2:d-2) ./ cumprod(2:2:d-1)...] ./ (1:2:d)
+    c /= sum(c)
+    g = Polynomial(c)
+    gprime = derivative(g)
+    g, gprime
+end
+
+"""
+```julia
+koslofftalezertransformation(α)
+```
+
+# Arguments:
+- `α`: polynomial degree of truncated Taylor series expansion of arcsin(s).
+
+# Returns:
+The Kosloff and Tal-Ezer conformal map derived from the inverse sine function.
+
+# Example:
+```jldoctest
+# koslofftalezertransformation conformal map
+numberquadraturepoints = 5;
+g, gprime = LFAToolkit.koslofftalezertransformation(0.95);
+
+# verify
+@assert g.(LinRange(-1,1,5)) ≈ [-1.0, -0.39494881426787537, 0.0, 0.39494881426787537, 1.0]
+
+# output
+
+```
+"""
+function koslofftalezertransformation(α)
+    g(s) = asin(α * s) / asin(α)
+    gprime(s) = α / (asin(α) * sqrt(1 - (α * s)^2))
+    g, gprime
+end
+
+"""
+```julia
+haletrefethenstriptransformation(ρ)
+```
+
+# Arguments:
+- `ρ`: sum of the semiminor and semimajor axis
+
+# Returns:
+The Hale and Trefethen strip transformation
+
+# Example:
+```jldoctest
+# haletrefethenstriptransformation conformal map
+g, gprime = haletrefethenstriptransformation(1.4);
+
+# verify
+@assert g.(LinRange(-1,1,5)) ≈ [-1.0, -0.36812132798370184, 0.0, 0.36812132798370184, 1.0]
+
+# output
+
+```
+"""
+function haletrefethenstriptransformation(ρ)
+    τ = π / log(ρ)
+    d = 0.5 + 1 / (exp(τ * π) + 1)
+    π2 = π / 2
+    # Unscaled functions of u
+    gu(u) = log(1 + exp(-τ * (π2 + u))) - log(1 + exp(-τ * (π2 - u))) + d * τ * u
+    gprimeu(u) = 1 / (exp(τ * (π2 + u)) + 1) + 1 / (exp(τ * (π2 - u)) + 1) - d
+    # Normalizing factor and scaled functions of s
+    C = 1 / gu(π / 2)
+    g(s) = C * gu(asin(s))
+    gprime(s) = -τ * C / sqrt(1 - s^2) * gprimeu(asin(s))
+    g, gprime
+end
+
+"""
+```julia
+transformquadrature(points, weights, mapping)
+```
+
+# Arguments:
+- `points`:  number quadrature points
+- `weights`: boolean flag indicating if quadrature weights are desired
+- `mapping`: choice of conformal map
+
+# Returns:
+Transformed quadrature by applying a smooth mapping = (g, gprime) from the original domain.
+
+# Example:
+```jldoctest
+# generate transformed quadrature points, weights with choice of conformal map
+points = 5;
+mapping = haletrefethenstriptransformation(1.4);
+mpoints, mweights = LFAToolkit.transformquadrature(points, true, mapping);
```

Comment:
You're literally applying the mapping to `points`, which is just the integer 5. You probably intended to apply it to an array of points between -1 and 1. This is the source of the error you're observing.